gateway/gateway.go

```diff
@@ -1053,34 +1053,34 @@ func (gateway *HandleT) StartWebHandler() {
 	gateway.logger.Infof("Starting in %d", webPort)
 	srvMux := mux.NewRouter()
 	srvMux.Use(headerMiddleware)
-	srvMux.HandleFunc("/v1/batch", gateway.stat(gateway.webBatchHandler))
-	srvMux.HandleFunc("/v1/identify", gateway.stat(gateway.webIdentifyHandler))
-	srvMux.HandleFunc("/v1/track", gateway.stat(gateway.webTrackHandler))
-	srvMux.HandleFunc("/v1/page", gateway.stat(gateway.webPageHandler))
-	srvMux.HandleFunc("/v1/screen", gateway.stat(gateway.webScreenHandler))
-	srvMux.HandleFunc("/v1/alias", gateway.stat(gateway.webAliasHandler))
-	srvMux.HandleFunc("/v1/merge", gateway.stat(gateway.webMergeHandler))
-	srvMux.HandleFunc("/v1/group", gateway.stat(gateway.webGroupHandler))
-	srvMux.HandleFunc("/health", gateway.healthHandler)
-	srvMux.HandleFunc("/v1/import", gateway.stat(gateway.webImportHandler))
-	srvMux.HandleFunc("/", gateway.healthHandler)
-	srvMux.HandleFunc("/pixel/v1/track", gateway.stat(gateway.pixelTrackHandler))
-	srvMux.HandleFunc("/pixel/v1/page", gateway.stat(gateway.pixelPageHandler))
-	srvMux.HandleFunc("/version", gateway.versionHandler)
-	srvMux.HandleFunc("/v1/webhook", gateway.stat(gateway.webhookHandler.RequestHandler))
-	srvMux.HandleFunc("/beacon/v1/batch", gateway.stat(gateway.beaconBatchHandler))
+	srvMux.HandleFunc("/v1/batch", gateway.stat(gateway.webBatchHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/identify", gateway.stat(gateway.webIdentifyHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/track", gateway.stat(gateway.webTrackHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/page", gateway.stat(gateway.webPageHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/screen", gateway.stat(gateway.webScreenHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/alias", gateway.stat(gateway.webAliasHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/merge", gateway.stat(gateway.webMergeHandler)).Methods("POST")
+	srvMux.HandleFunc("/v1/group", gateway.stat(gateway.webGroupHandler)).Methods("POST")
+	srvMux.HandleFunc("/health", gateway.healthHandler).Methods("GET")
+	srvMux.HandleFunc("/v1/import", gateway.stat(gateway.webImportHandler)).Methods("POST")
+	srvMux.HandleFunc("/", gateway.healthHandler).Methods("GET")
+	srvMux.HandleFunc("/pixel/v1/track", gateway.stat(gateway.pixelTrackHandler)).Methods("GET")
+	srvMux.HandleFunc("/pixel/v1/page", gateway.stat(gateway.pixelPageHandler)).Methods("GET")
+	srvMux.HandleFunc("/version", gateway.versionHandler).Methods("GET")
+	srvMux.HandleFunc("/v1/webhook", gateway.stat(gateway.webhookHandler.RequestHandler)).Methods("POST")
```

Comment:
We should support POST and GET for the webhooks. Some of the destinations might send a GET request only

Reply:
done